README.md

```diff
@@ -226,6 +227,36 @@ if (!userInfo) {
 }
 ```
 
+## Routing strategies
+**!** Routing is **optional** strategy for token handling. Instead you can use [popup](#tokengetwithpopupoptions) or [sign widget](https://github.com/okta/okta-signin-widget).
+
+Handling and defining of redirect URI depends on what route strategy you use in your project.
+
+### Path-based routing strategy
+
+In case of using path-base `/` routing strategy it is possible to define redirect URI with a dedicated callback route.
+
+#### The flow for okta-auth-js with path-based router:
+1. Right before redirect, save the route you are on (we recommend sessionStorage)
+2. Do the redirect to okta
+3. Redirect back to a dedicated route
+4. Call *parseFromUrl()*, retrieve tokens, add to `tokenManager`
+5. Read saved route and redirect to it
+
+### Hash-based routing strategy
+
+According to the OAuth 2.0 spec the redirect URI "MUST NOT contain a fragment component": <https://tools.ietf.org/html/rfc6749#section-3.1.2>
+So in case of using hash-based `#` strategy and OAuth 2.0, the redirect URI can be defined only like a base url, without any specific rout.
+That's mean that hash-based router will receive the redirect callback on the main / default route. So we recommend to define the logic that will parse redirect url at the very beginning of your app.
+
+Additionally if using hash routing, we recommend to use PKCE and responseMode query (which is the default for PKCE). Using implicit flow, with tokens in the hash could cause unpredictable results since hash routers like to rewrite the fragment.
+
+#### The flow for okta-auth-js with hash-based router:
```

Comment:
as mentioned, routing is optional. This strategy (for hash-based router) is the same strategy as a single page application with no routing. Let's move this strategy up to the top. So we say routing is optional, here's the way it works with a single page. Then we describe how to do callback with a dedicated route. Then for hash-based, we explain the stuff about the spec and why even though you are using routing, it will not be used for the login callback. Then we link back to the section on SPA strategy with no routing.